File: output/typescript_rest_sdk/client_sdk/async_client.ts
import { Pet, NewPet, Category, Error } from './types';
import { , ClientConfig } from './client';

/**
 * Pet Store API TypeScript SDK Async Client
 * 
 * A sample API that uses a pet store as an example
 * 
 * Provides Promise-based async operations (TypeScript/JavaScript is naturally async)
 */
export class Async {
  private client: ;

  constructor(config: ClientConfig) {
    this.client = new (config);
  }


  /**
   * Returns a list of all pets in the store (Async)
   * @param  Maximum number of pets to return
   * @param  Filter pets by tag
   */
  async listpetsAsync(?: string, ?: string): Promise<> {
    // In TypeScript/JavaScript, all operations are naturally async
    return this.client.listpets(, );
  }

  /**
   * Creates a new pet in the store (Async)
   */
  async createpetAsync(): Promise<> {
    // In TypeScript/JavaScript, all operations are naturally async
    return this.client.createpet();
  }

  /**
   * Returns a single pet by its ID (Async)
   * @param  ID of the pet to retrieve
   */
  async getpetAsync(: string): Promise<> {
    // In TypeScript/JavaScript, all operations are naturally async
    return this.client.getpet();
  }

  /**
   * Updates an existing pet (Async)
   * @param  ID of the pet to update
   */
  async updatepetAsync(: string): Promise<> {
    // In TypeScript/JavaScript, all operations are naturally async
    return this.client.updatepet();
  }

  /**
   * Deletes a pet from the store (Async)
   * @param  ID of the pet to delete
   */
  async deletepetAsync(: string): Promise<> {
    // In TypeScript/JavaScript, all operations are naturally async
    return this.client.deletepet();
  }

}

File: output/go_rest_sdk/client_sdk/async_client.go
package client

import (
	"context"
	"fmt"
)

// PetStoreApiclientAsyncClient provides async operations for Pet Store API
// In Go, this is typically handled through goroutines and channels
type PetStoreApiclientAsyncClient struct {
	client *PetStoreApiclient
}

// NewPetStoreApiclientAsync creates a new async client
func NewPetStoreApiclientAsync(config ClientConfig) *PetStoreApiclientAsyncClient {
	return &PetStoreApiclientAsyncClient{
		client: NewPetStoreApiclient(config),
	}
}

// ListpetsAsync Returns a list of all pets in the store (Async)
func (c *Pet Store APIClientAsyncClient) ListpetsAsync(ctx context.Context, limit string, tag string) <-chan ListpetsResult {
	resultChan := make(chan ListpetsResult, 1)
	
	go func() {
		defer close(resultChan)
		
		result, err := c.client.Listpets(ctx, limit, tag)
		resultChan <- ListpetsResult{
			Data:  result,
			Error: err,
		}
	}()
	
	return resultChan
}

// ListpetsResult holds the result of Listpets operation
type ListpetsResult struct {
	Data  
	Error error
}

// CreatepetAsync Creates a new pet in the store (Async)
func (c *Pet Store APIClientAsyncClient) CreatepetAsync(ctx context.Context) <-chan CreatepetResult {
	resultChan := make(chan CreatepetResult, 1)
	
	go func() {
		defer close(resultChan)
		
		result, err := c.client.Createpet(ctx)
		resultChan <- CreatepetResult{
			Data:  result,
			Error: err,
		}
	}()
	
	return resultChan
}

// CreatepetResult holds the result of Createpet operation
type CreatepetResult struct {
	Data  
	Error error
}

// GetpetAsync Returns a single pet by its ID (Async)
func (c *Pet Store APIClientAsyncClient) GetpetAsync(ctx context.Context, petid string) <-chan GetpetResult {
	resultChan := make(chan GetpetResult, 1)
	
	go func() {
		defer close(resultChan)
		
		result, err := c.client.Getpet(ctx, petid)
		resultChan <- GetpetResult{
			Data:  result,
			Error: err,
		}
	}()
	
	return resultChan
}

// GetpetResult holds the result of Getpet operation
type GetpetResult struct {
	Data  
	Error error
}

// UpdatepetAsync Updates an existing pet (Async)
func (c *Pet Store APIClientAsyncClient) UpdatepetAsync(ctx context.Context, petid string) <-chan UpdatepetResult {
	resultChan := make(chan UpdatepetResult, 1)
	
	go func() {
		defer close(resultChan)
		
		result, err := c.client.Updatepet(ctx, petid)
		resultChan <- UpdatepetResult{
			Data:  result,
			Error: err,
		}
	}()
	
	return resultChan
}

// UpdatepetResult holds the result of Updatepet operation
type UpdatepetResult struct {
	Data  
	Error error
}

// DeletepetAsync Deletes a pet from the store (Async)
func (c *Pet Store APIClientAsyncClient) DeletepetAsync(ctx context.Context, petid string) <-chan DeletepetResult {
	resultChan := make(chan DeletepetResult, 1)
	
	go func() {
		defer close(resultChan)
		
		result, err := c.client.Deletepet(ctx, petid)
		resultChan <- DeletepetResult{
			Data:  result,
			Error: err,
		}
	}()
	
	return resultChan
}

// DeletepetResult holds the result of Deletepet operation
type DeletepetResult struct {
	Data  
	Error error
}

